File: tests/integration/construct_component_error_apply/nodejs/component.ts
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

import * as pulumi from "@pulumi/pulumi";

interface ComponentArgs {
    foo: pulumi.Input<string>;
}

export class Component extends pulumi.ComponentResource {
    public readonly foo!: pulumi.Output<string>;

    constructor(name: string, args: ComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        const inputs: any = {};
        inputs["foo"] = args.foo;

        super("testcomponent:index:Component", name, inputs, opts, true);
    }
}

File: tests/integration/construct_component_unknown/python/component.py
# Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

from typing import Optional

import pulumi

class Component(pulumi.ComponentResource):
    def __init__(self, name: str, args: pulumi.Inputs, opts: Optional[pulumi.ResourceOptions] = None):
        super().__init__("testcomponent:index:Component", name, args, opts, True)
